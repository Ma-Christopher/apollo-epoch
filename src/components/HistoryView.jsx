import * as React from 'react';

// export interface HistoryViewProps {}

const HistoryView = () => {
  return (
    <div className="history-view">
      <h1>Queries</h1>
      <div className="query-card">
        <p>Query 1</p>
      </div>
      <div className="query-card">
        <p>Query 2</p>
      </div>
      <div className="query-card">
        <p>Query 3</p>
      </div>
    </div>
  );
};

export default HistoryView;
