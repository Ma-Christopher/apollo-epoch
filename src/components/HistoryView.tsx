import * as React from 'react';

export interface HistoryViewProps {}

const HistoryView: React.FC<HistoryViewProps> = () => {
  return <div>List out the queries performed here</div>;
};

export default HistoryView;
