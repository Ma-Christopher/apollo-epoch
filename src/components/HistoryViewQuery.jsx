/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { createMuiTheme, ThemeProvider } from '@material-ui/core/styles';
import networkIcon from '../static/flux_capacitor.png';
import { initiateEpochShift } from '../store/entities/apollo';

const epochTheme = createMuiTheme({ palette: { primary: { main: '#20909f' } } });

const HistoryViewQuery = ({ timelineObj, onClick, active }) => {
  const { id, name, isNetwork } = timelineObj;
  const dispatch = useDispatch();

  const epochShift = (apolloActionId) => {
    dispatch(initiateEpochShift({ apolloActionId }));
  };

  return (
    <div
      className={`query-card ${active && 'active-query'}`}
      onClick={onClick}
      active={active}
      id={id}
    >
      <div className="query-card-text">
        <div>{`${id}: ${name}`}</div>
        <div className="network-icon-and-jump">
          {isNetwork && <img src={networkIcon} alt="network" className="network-icon" />}
          <ThemeProvider theme={epochTheme}>
            {id[0] !== 'F' && (
              <div className="button-switcher-container">
                <div className="button-switcher">
                  <div className="time-stamp">
                    <p>No Time</p>
                  </div>
                  <div
                    className="jump-button"
                    onClick={() => {
                      epochShift(id);
                    }}
                  >
                    <p>Jump</p>
                  </div>
                </div>
              </div>
            )}
          </ThemeProvider>
        </div>
      </div>
    </div>
  );
};

HistoryViewQuery.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  timelineObj: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired,
  active: PropTypes.bool.isRequired,
};

export default HistoryViewQuery;
