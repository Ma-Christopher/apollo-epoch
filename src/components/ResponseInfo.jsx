import * as React from 'react';

// export interface ResponseInfoProps {}

const ResponseInfo = () => {
  return <div>ResponseInfo here</div>;
};

export default ResponseInfo;
