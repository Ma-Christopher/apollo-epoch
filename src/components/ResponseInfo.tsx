import * as React from 'react';

export interface ResponseInfoProps {}

const ResponseInfo: React.FC<ResponseInfoProps> = () => {
  return <div>ResponseInfo here</div>;
};

export default ResponseInfo;
