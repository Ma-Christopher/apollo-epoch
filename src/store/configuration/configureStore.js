import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import rootReducer from './rootReducer';
import initializePort from '../middleware/createBackgroundPort';

export default function () {
  return configureStore({
    reducer: rootReducer,
    middleware: [
      ...getDefaultMiddleware(), // returns THUNK and compose w/ DevTools
      initializePort,
    ],
  });
}
