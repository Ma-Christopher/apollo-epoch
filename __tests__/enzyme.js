import React from 'react';
import { configure, shallow, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { Provider, useSelector, useDispatch } from 'react-redux';
import configureStore from 'redux-mock-store';
import StateValueDisplay from '../src/components/StateValueDisplay';
import HistoryView from '../src/components/HistoryView';
import StateSidebar from '../src/components/StateSidebar';
import ResponseInfo from '../src/components/ResponseInfo';
import HistoryViewQuery from '../src/components/HistoryViewQuery';
// import QueryInfo from '../src/components/QueryInfo';

configure({ adapter: new Adapter() });
const mockStore = configureStore([]);

describe('React unit tests', () => {
  describe('StateValueDisplay', () => {
    let wrapper;

    const props = {
      stateValue: {
        testVal: 1,
        testString: 'hello',
      },
    };

    beforeAll(() => {
      wrapper = shallow(<StateValueDisplay {...props} />);
    });

    it('It should render without errors', () => {
      const component = toJson(wrapper);
      expect(component).toMatchSnapshot();
    });

    it('It contains a div with "state-value" as the class type', () => {
      expect(wrapper.type()).toEqual('div');
      const component = wrapper.find('.state-value');
      expect(component.length).toBe(1);
    });
  });

  describe('StateSideBar', () => {
    let wrapper;
    const props = {
      keyList: ['1', '2', '3'],
      onClick: () => {},
    };

    beforeAll(() => {
      wrapper = shallow(<StateSidebar {...props} />);
    });

    it('It should render without errors', () => {
      const component = toJson(wrapper);
      expect(component).toMatchSnapshot();
    });

    it('It should render a "key-display" div for each key in the props', () => {
      const keyDisplays = wrapper.find('.key-display');
      expect(keyDisplays.length).toBe(3);
    });
  });

  describe('ResponseInfo', () => {
    let wrapper;
    const props = {
      response: { res: 'test' },
    };

    beforeAll(() => {
      wrapper = shallow(<ResponseInfo {...props} />);
    });

    it('It should render without errors', () => {
      const component = toJson(wrapper);
      expect(component).toMatchSnapshot();
    });

    it('It should contain a single div with the "response-info" class', () => {
      const responseInfo = wrapper.find('.response-info');
      expect(responseInfo.length).toBe(1);
    });

    it('It should display "No Response to Render" if the associated query does not have a response', () => {
      wrapper = shallow(<ResponseInfo response={null} />);
      expect(wrapper.find('h2').text()).toMatch('No Response to Render');
    });
  });

  xdescribe('HistoryView', () => {
    let store;
    let wrapper;
    const apollo = {
      hasDunderApollo: false,
      loadingApollo: false,
      activeQuery: {},
      prevQuery: {},
      chromeTabId: '',
      graphQlUri: '',
      queryIds: [],
      queries: {},
      queryIdCounter: 1,
      mutationIds: [],
      mutations: {},
      mutationIdCounter: 1,
      manualFetches: {}, // store manual cacheFetches in Timeline
      manualFetchIds: [],
      fetchCounter: 0,
      timeline: [], // an ordered list of query and mutation Ids
      typeNameDocumentCache: {},
      networkHoldingRoom: {},
    };

    beforeAll(() => {
      store = mockStore({ apollo });

      wrapper = render(
        <Provider store={store}>
          <HistoryView />
        </Provider>
      );
    });

    it('It should render with given state from Redux store', () => {
      const component = toJson(wrapper);
      expect(component).toMatchSnapshot();
    });
  });
});
