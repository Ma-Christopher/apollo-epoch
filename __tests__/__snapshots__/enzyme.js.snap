// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit tests ResponseInfo It should render without errors 1`] = `
<div
  className="response-info"
>
  <t
    collapseStringsAfterLength={false}
    collapsed={false}
    defaultValue={null}
    displayDataTypes={false}
    displayObjectSize={false}
    enableClipboard={false}
    groupArraysAfterLength={100}
    iconStyle="triangle"
    indentWidth={2}
    name={false}
    onAdd={false}
    onDelete={false}
    onEdit={false}
    onSelect={false}
    shouldCollapse={false}
    sortKeys={false}
    src={
      Object {
        "res": "test",
      }
    }
    style={Object {}}
    theme={
      Object {
        "base00": "#0a090c",
        "base01": "red",
        "base02": "hsl(0, 0%, 35%)",
        "base03": "blue",
        "base04": "purple",
        "base05": "hsl(0, 100%, 75%)",
        "base06": "orange",
        "base07": "white",
        "base08": "white",
        "base09": "rgb(155, 187, 220)",
        "base0A": "hsl(0, 100%, 75%)",
        "base0B": "green",
        "base0C": "white",
        "base0D": "rgb(231, 194, 112)",
        "base0E": "#c678dd",
        "base0F": "rgb(53, 212, 199)",
      }
    }
    validationMessage="Validation Error"
  />
</div>
`;

exports[`React unit tests StateSideBar It should render without errors 1`] = `
<div
  className="state-sidebar"
>
  <div
    className="keylist-wrapper"
  >
    <div
      className="key-display "
      id="1"
      key="key0"
      onClick={[Function]}
    >
      1
    </div>
    <div
      className="key-display "
      id="2"
      key="key1"
      onClick={[Function]}
    >
      2
    </div>
    <div
      className="key-display "
      id="3"
      key="key2"
      onClick={[Function]}
    >
      3
    </div>
  </div>
</div>
`;

exports[`React unit tests StateValueDisplay It should render without errors 1`] = `
<div
  className="state-value"
>
  <t
    collapseStringsAfterLength={false}
    collapsed={false}
    defaultValue={null}
    displayDataTypes={false}
    displayObjectSize={false}
    enableClipboard={false}
    groupArraysAfterLength={100}
    iconStyle="triangle"
    indentWidth={2}
    name={false}
    onAdd={false}
    onDelete={false}
    onEdit={false}
    onSelect={false}
    shouldCollapse={false}
    sortKeys={false}
    src={
      Object {
        "testString": "hello",
        "testVal": 1,
      }
    }
    style={Object {}}
    theme={
      Object {
        "base00": "#0a090c",
        "base01": "red",
        "base02": "hsl(0, 0%, 35%)",
        "base03": "blue",
        "base04": "purple",
        "base05": "hsl(0, 100%, 75%)",
        "base06": "orange",
        "base07": "white",
        "base08": "white",
        "base09": "rgb(155, 187, 220)",
        "base0A": "hsl(0, 100%, 75%)",
        "base0B": "green",
        "base0C": "white",
        "base0D": "rgb(231, 194, 112)",
        "base0E": "#c678dd",
        "base0F": "rgb(53, 212, 199)",
      }
    }
    validationMessage="Validation Error"
  />
</div>
`;
